app/models/ability.rb

```diff
@@ -0,0 +1,38 @@
+class Ability
+  include CanCan::Ability
+
+  def initialize(user)
+    # Define abilities for the user here. For example:
+    can :read, Recipe if recipe.public?
+
+    return unless user.present?
+
+    can :manage, :all, user_id: user.id
+
+    return unless user.role == 'admin'
+
+    can :manage, :all
+    #   return unless user.present?
```

Comment:
Please I suggest that we can refactor the roles for  a regular user  and that of an admin
into something like this

```suggestion
     user ||= User.new # guest user (not logged in)
    if user.role == 'admin'
      can :manage, :all
    else
      can :read, Recipe do |recipe|
        recipe.user == user || recipe.public
      end

      can %i[update destroy], Recipe do |recipe|
        recipe.user == user
      end

      can :read, Food
      can :destroy, Food do |food|
        food.user.id == user.id
      end
      can %i[create], :all
    end
```

Reply:
Thank you @assadounto for your kind suggestions, I have implemented the required changes 😄